gcloud-java-compute/src/main/java/com/google/gcloud/compute/ComputeImpl.java

```diff
@@ -914,6 +933,100 @@ public Operation delete(final AddressId addressId, OperationOption... options) {
     }
   }
 
+  @Override
+  public Operation create(SnapshotInfo snapshot, OperationOption... options) {
+    final SnapshotInfo completeSnapshot = snapshot.setProjectId(options().projectId());
+    final Map<ComputeRpc.Option, ?> optionsMap = optionMap(options);
+    try {
+      com.google.api.services.compute.model.Operation answer =
+          runWithRetries(new Callable<com.google.api.services.compute.model.Operation>() {
+            @Override
+            public com.google.api.services.compute.model.Operation call() {
+              return computeRpc.createSnapshot(completeSnapshot.sourceDisk().zone(),
+                  completeSnapshot.sourceDisk().disk(), completeSnapshot.snapshotId().snapshot(),
+                  completeSnapshot.description(), optionsMap);
+            }
+          }, options().retryParams(), EXCEPTION_HANDLER);
+      return answer == null ? null : Operation.fromPb(this, answer);
+    } catch (RetryHelper.RetryHelperException e) {
+      throw ComputeException.translateAndThrow(e);
+    }
+  }
+
+  @Override
+  public Snapshot getSnapshot(final String snapshot, SnapshotOption... options) {
+    final Map<ComputeRpc.Option, ?> optionsMap = optionMap(options);
+    try {
+      com.google.api.services.compute.model.Snapshot answer =
+          runWithRetries(new Callable<com.google.api.services.compute.model.Snapshot>() {
+            @Override
+            public com.google.api.services.compute.model.Snapshot call() {
+              return computeRpc.getSnapshot(snapshot, optionsMap);
+            }
+          }, options().retryParams(), EXCEPTION_HANDLER);
+      return answer == null ? null : Snapshot.fromPb(this, answer);
+    } catch (RetryHelper.RetryHelperException e) {
+      throw ComputeException.translateAndThrow(e);
+    }
+  }
+
+  @Override
+  public Page<Snapshot> listSnapshots(SnapshotListOption... options) {
+    return listSnapshots(options(), optionMap(options));
+  }
+
+  private static Page<Snapshot> listSnapshots(final ComputeOptions serviceOptions,
```

Comment:
Codacy uses [PMD](http://pmd.sourceforge.net/) under the hood, this is probably a PMD bug. I will forward the issue.